std/experimental/allocator/package.d

```diff
@@ -1121,40 +1121,49 @@ propagates the exception.
 */
 auto make(T, Allocator, A...)(auto ref Allocator alloc, auto ref A args)
 {
-    import std.algorithm.comparison : max;
-    import std.conv : emplace, emplaceRef;
-    auto m = alloc.allocate(max(stateSize!T, 1));
-    if (!m.ptr) return null;
-
-    // make can only be @safe if emplace or emplaceRef is `pure`
-    auto construct()
+    static if (!is(T == class) && A.length == 0 && isInitAllZeroBits!T
+        && is(typeof(alloc.allocateZeroed(size_t.max))))
     {
-        static if (is(T == class)) return emplace!T(m, args);
-        else
-        {
-            // Assume cast is safe as allocation succeeded for `stateSize!T`
-            auto p = () @trusted { return cast(T*) m.ptr; }();
-            emplaceRef(*p, args);
-            return p;
-        }
+        auto m = alloc.allocateZeroed(T.sizeof ? T.sizeof : 1);
+        return (() @trusted => cast(T*) m.ptr)();
```

Comment:
We should probably add a test that `isInitAllZeroBits` returns false when for `@disable this`.
(there's none atm)

Reply:
For instance, in `uninitializedFillDefault` we want to know what's in `T.init` and we don't care if the default constructor is disabled.

Reply:
(Check added to condition)